src/main/java/core/basesyntax/service/ReportCreatorService.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.service;
+
+import java.util.List;
+import java.util.Map;
+
+public interface ReportCreatorService {
+    List<String> createReport(Map<String, Integer> products);
```

Comment:
if the returned report is of type String, your WriterService will only have one responsibility and will be much smaller. Also, the write process will be performed in a single iteration